apps/web/src/components/Profile/index.tsx

```diff
@@ -37,6 +44,38 @@ const ViewProfile: NextPage = () => {
     skip: !handle
   });
 
+  const profile = data?.profile;
+  const [following, setFollowing] = useState<boolean | null>(null);
+  const [showFollowModal, setShowFollowModal] = useState(false);
+
+  // workaround for that profile.isFollowedByMe == true when signed out
+  const isFollowedByMe = !!currentProfile && !!profile?.isFollowedByMe;
+
+  const followType = profile?.followModule?.__typename;
+
+  const initState = following == null;
+  // profile is not defined until the second render
+  if (initState && profile) {
+    const canFollow = followType !== 'RevertFollowModuleSettings' && !isFollowedByMe;
+    if (showFollowDialog && canFollow) {
+      setShowFollowModal(true);
+    }
+    setFollowing(isFollowedByMe);
+  }
+
+  // profile changes when user selects a new profile from search box
+  useEffect(() => {
+    if (profile != undefined) {
```

Comment:
```suggestion
    if (profile !== undefined) {
```

will `!profile` work here?

Reply:
hmm, not sure why I wrote `if (profile != undefined)` and not simply `if (profile)`
i have updated the code